inc/blocks/namespace.php

```diff
@@ -157,6 +159,86 @@ function on_save_post( int $post_ID, WP_Post $post, bool $update ) : void {
 	}
 }
 
+/**
+ * Synchronise any XBs on the widget page with a shadow post type.
+ *
+ * @param instance $instance Widget Object.
+ * @param new_instance $new_instance Widget Object.
+ * @param old_instance $old_instance Widget Object
+ * @param widget $widget Current Widget Instance
+ * @return object
+ */
+function on_widgets_save( $instance, $new_instance, $old_instance, $widget ) {
+
+	// Scan for XBs in the post content.
+	$blocks = parse_blocks( $new_instance['content'] );
```

Comment:
I think you can just check `$instance['content']` here as `$instance` has already been through the widget class's `update` method.